prov/rxd/src/rxd_cq.c

```diff
@@ -311,7 +315,7 @@ static void rxd_handle_data(struct rxd_ep *ep, struct fi_cq_msg_entry *comp,
 		return;
 	}
 
-	if (pkt_seg_no == rx_entry->next_seg_no) {
+	if (pkt_seg_no == rx_entry->next_seg_no || rxd_env.ooo_rdm) {
```

Comment:
If we're going to support OOO messages, you can consider adding selective acks, rather than defaulting to a go back n retransmission algorithm.

Reply:
I'll note that for a future addition for the mess that is the ACK and windowing problem. Right now the only OOO messages it can support is reliable ones, so there will be no retransmission anyway

Reply:
Yes, this was meant as a future consideration, not a change request for this PR.  :)